tool/tctl/common/autoupdate_command.go

```diff
@@ -0,0 +1,239 @@
+
+	clientToolsCmd := autoUpdateCmd.Command("client-tools", "Client tools auto update commands.")
+
+	c.setCmd = clientToolsCmd.Command("set", "Sets client tools auto update configuration.")
+	c.setCmd.Flag("mode", "Defines the mode to enable or disable tools auto update in cluster.").EnumVar(&c.mode, "enabled", "disabled", "on", "off")
+	c.setCmd.Flag("target-version", "Defines client tools target version required to be updated.").StringVar(&c.toolsTargetVersion)
+
+	c.getCmd = clientToolsCmd.Command("get", "Receive tools auto update target version.")
+	c.getCmd.Flag("proxy", "Address of the Teleport proxy. When defined this address going to be used for requesting target version for auto update.").StringVar(&c.proxy)
+	c.getCmd.Flag("format", "Output format: 'yaml' or 'json'").Default(teleport.YAML).StringVar(&c.format)
+
+	if c.stdout == nil {
+		c.stdout = os.Stdout
+	}
+}
+
+// TryRun takes the CLI command as an argument and executes it.
+func (c *AutoUpdateCommand) TryRun(ctx context.Context, cmd string, client *authclient.Client) (match bool, err error) {
+	switch cmd {
+	case c.setCmd.FullCommand():
+		err = c.Set(ctx, client)
+	case c.getCmd.FullCommand():
+		err = c.Get(ctx, client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+// Set creates or updates AutoUpdateConfig and AutoUpdateVersion resources with specified parameters.
+func (c *AutoUpdateCommand) Set(ctx context.Context, client *authclient.Client) error {
+	if c.mode != "" {
+		if err := c.setAutoUpdateConfig(ctx, client); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	if c.toolsTargetVersion != "" {
+		if err := c.setAutoUpdateVersion(ctx, client); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+// Get makes request to fetch tools autoupdate version, if proxy flag is not set
+// authorized handler should be used.
+func (c *AutoUpdateCommand) Get(ctx context.Context, client *authclient.Client) error {
+	response, err := c.get(ctx, client)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	switch c.format {
+	case teleport.JSON:
+		if err := utils.WriteJSON(c.stdout, response); err != nil {
+			return trace.Wrap(err)
+		}
+	case teleport.YAML:
+		if err := utils.WriteYAML(c.stdout, response); err != nil {
+			return trace.Wrap(err)
+		}
+	default:
+		return trace.BadParameter("unsupported output format %s, supported values are %s and %s", c.format, teleport.JSON, teleport.YAML)
+	}
+
+	return nil
+}
+
+func (c *AutoUpdateCommand) setAutoUpdateConfig(ctx context.Context, client *authclient.Client) error {
+	configExists := true
+	config, err := client.GetAutoUpdateConfig(ctx)
+	if trace.IsNotFound(err) {
+		configExists = false
+		if config, err = autoupdate.NewAutoUpdateConfig(&autoupdatev1pb.AutoUpdateConfigSpec{}); err != nil {
+			return trace.Wrap(err)
+		}
+	} else if err != nil {
+		return trace.Wrap(err)
+	}
+	switch c.mode {
+	case "on":
+		c.mode = autoupdate.ToolsUpdateModeEnabled
+	case "off":
```

Comment:
if we don't set anything we don't have to modify the mode, thats why we have no default value, otherwise command like `tctl autoupdate client-tools set --target-version 16.4.0` going to disable autoupdates

Reply:
sorry I was switching between code and comments and again misinterpret your original comment, I will drop `enabled` and `disabled` from enums and leave only `on/off`

Reply:
Maybe I'm misunderstanding, but reading https://github.com/gravitational/teleport/pull/47692#discussion_r1813227463, I think we want to support all four values from tctl, and translate "on" / "off" to "enabled" / "disabled," so they serve as shortcuts.